File: Shared/Blog.Contracts/V1/ApiRoutes.cs
namespace Blog.Contracts.V1
{
    /// <summary>
    /// Api routes constants.
    /// </summary>
    public static class ApiRoutes
    {
        /// <summary>
        /// The root.
        /// </summary>
        public const string Root = "api";

        /// <summary>
        /// The version.
        /// </summary>
        public const string Version = "v1";

        /// <summary>
        /// The base.
        /// </summary>
        public const string Base = Root + "/" + Version;

        /// <summary>
        /// Accounts controller routes constants.
        /// </summary>
        public static class AccountsController
        {
            /// <summary>
            /// The accounts.
            /// </summary>
            public const string Accounts = Base + "/accounts";

            /// <summary>
            /// The initialize.
            /// </summary>
            public const string Initialize = "initialize/{userId}";

            /// <summary>
            /// The get all users.
            /// </summary>
            public const string GetAllUsers = "get-all-users";

            /// <summary>
            /// The send confirmation email.
            /// </summary>
            public const string SendConfirmationEmail = "send-confirmation-email";

            /// <summary>
            /// The login.
            /// </summary>
            public const string Login = "login";

            /// <summary>
            /// The register.
            /// </summary>
            public const string Register ="register";

            /// <summary>
            /// The change password.
            /// </summary>
            public const string ChangePassword = "change-password";
        }

        /// <summary>
        /// Comments controller routes constants.
        /// </summary>
        public static class CommentsController
        {
            /// <summary>
            /// The comments.
            /// </summary>
            public const string Comments = Base + "/comments";

            /// <summary>
            /// The get comments by post.
            /// </summary>
            public const string GetCommentsByPost = "get-comments-by-post/{id}";

            /// <summary>
            /// The get comments by filter.
            /// </summary>
            public const string GetCommentsByFilter = "get-comments-by-filter";

            /// <summary>
            /// The get comment.
            /// </summary>
            public const string GetComment = "get-comment";

            /// <summary>
            /// The create comment.
            /// </summary>
            public const string CreateComment = "create";
        }

        /// <summary>
        /// Messages controller routes constants.
        /// </summary>
        public static class MessagesController
        {
            /// <summary>
            /// The messages.
            /// </summary>
            public const string Messages = Base + "/messages";

            /// <summary>
            /// The get recipient messages.
            /// </summary>
            public const string GetRecipientMessages = "get-recipient-messages/{recipientId}";

            /// <summary>
            /// The get sender messages.
            /// </summary>
            public const string GetSenderMessages = "get-sender-messages/{senderEmail}";

            /// <summary>
            /// The show.
            /// </summary>
            public const string Show = "show/{id}";
        }

        /// <summary>
        /// Posts controller routes constants.
        /// </summary>
        public static class PostsController
        {
            /// <summary>
            /// The posts.
            /// </summary>
            public const string Posts = Base + "/posts";

            /// <summary>
            /// The get posts.
            /// </summary>
            public const string GetPosts = "get-posts";

            /// <summary>
            /// The user posts.
            /// </summary>
            public const string UserPosts = "user-posts/{id}";

            /// <summary>
            /// The show.
            /// </summary>
            public const string Show = "show/{id}";

            /// <summary>
            /// The like post.
            /// </summary>
            public const string LikePost = "like/{id}";

            /// <summary>
            /// The dislike post.
            /// </summary>
            public const string DislikePost = "dislike/{id}";
        }

        /// <summary>
        /// Profile controller routes constants.
        /// </summary>
        public static class ProfileController
        {
            /// <summary>
            /// The profile.
            /// </summary>
            public const string Profile = Base + "/profile";
        }

        /// <summary>
        /// Tags controller routes constants.
        /// </summary>
        public static class TagsController
        {
            /// <summary>
            /// The tags.
            /// </summary>
            public const string Tags = Base + "/tags";

            /// <summary>
            /// The get tags.
            /// </summary>
            public const string GetTags = "get-tags";

            /// <summary>
            /// The get tags by filter.
            /// </summary>
            public const string GetTagsByFilter = "get-tags-by-filter";

            /// <summary>
            /// The get available tags.
            /// </summary>
            public const string GetAvailableTags = "get-available-tags/{postId}";

            /// <summary>
            /// The get tag.
            /// </summary>
            public const string GetTag = "get-tag";

            /// <summary>
            /// The create tag.
            /// </summary>
            public const string CreateTag = "create";
        }
    }
}

File: Shared/Blog.FSharp.Contracts/V1/ApiRoutes.fs
namespace Blog.FSharp.Contracts.V1

/// <summary>
/// API routes constants.
/// </summary>
module ApiRoutes =

    /// <summary>The root.</summary>
    let Root = "api"

    /// <summary>The version.</summary>
    let Version = "v1"

    /// <summary>The base route.</summary>
    let Base = Root + "/" + Version

    /// <summary>Accounts controller routes constants.</summary>
    module AccountsController =
        /// <summary>The accounts.</summary>
        let Accounts = Base + "/accounts"
        /// <summary>The initialize route.</summary>
        let Initialize = "initialize/{userId}"
        /// <summary>Get all users.</summary>
        let GetAllUsers = "get-all-users"
        /// <summary>Send confirmation email.</summary>
        let SendConfirmationEmail = "send-confirmation-email"
        /// <summary>Login route.</summary>
        let Login = "login"
        /// <summary>Register route.</summary>
        let Register = "register"
        /// <summary>Change password route.</summary>
        let ChangePassword = "change-password"
        /// <summary>Users activity route.</summary>
        let UsersActivity = "users-activity"

    /// <summary>Comments controller routes constants.</summary>
    module CommentsController =
        /// <summary>Comments route.</summary>
        let Comments = Base + "/comments"
        /// <summary>Get comments by post route.</summary>
        let GetCommentsByPost = "get-comments-by-post/{id:int}"
        /// <summary>Get comments by filter route.</summary>
        let GetCommentsByFilter = "get-comments-by-filter"
        /// <summary>Get comment route.</summary>
        let GetComment = "get-comment"
        /// <summary>Create comment route.</summary>
        let CreateComment = "create"
        /// <summary>Edit comment route.</summary>
        let EditComment = "{id:int}"
        /// <summary>Delete comment route.</summary>
        let DeleteComment = "{id:int}"
        /// <summary>Comments activity route.</summary>
        let CommentsActivity = "comments-activity"

    /// <summary>Messages controller routes constants.</summary>
    module MessagesController =
        /// <summary>Messages route.</summary>
        let Messages = Base + "/messages"
        /// <summary>Get recipient messages route.</summary>
        let GetRecipientMessages = "get-recipient-messages/{recipientId}"
        /// <summary>Get sender messages route.</summary>
        let GetSenderMessages = "get-sender-messages/{senderEmail}"
        /// <summary>Show route.</summary>
        let Show = "show/{id:int}"

    /// <summary>Posts controller routes constants.</summary>
    module PostsController =
        /// <summary>Posts route.</summary>
        let Posts = Base + "/posts"
        /// <summary>Get posts route.</summary>
        let GetPosts = "get-posts"
        /// <summary>User posts route.</summary>
        let UserPosts = "user-posts/{id}"
        /// <summary>Show route.</summary>
        let Show = "show/{id:int}"
        /// <summary>Like post route.</summary>
        let LikePost = "like/{id:int}"
        /// <summary>Dislike post route.</summary>
        let DislikePost = "dislike/{id:int}"
        /// <summary>Posts activity route.</summary>
        let PostsActivity = "posts-activity"

    /// <summary>Profile controller routes constants.</summary>
    module ProfileController =
        /// <summary>Profile route.</summary>
        let Profile = Base + "/profile"

    /// <summary>Tags controller routes constants.</summary>
    module TagsController =
        /// <summary>Tags route.</summary>
        let Tags = Base + "/tags"
        /// <summary>Get tags.</summary>
        let GetTags = "get-tags"
        /// <summary>Get tags by filter route.</summary>
        let GetTagsByFilter = "get-tags-by-filter"
        /// <summary>Get available tags route.</summary>
        let GetAvailableTags = "get-available-tags/{postId:int}"
        /// <summary>Get tag route.</summary>
        let GetTag = "get-tag/{id:int}"
        /// <summary>Create tag route.</summary>
        let CreateTag = "create"
        /// <summary>Tags activity route.</summary>
        let TagsActivity = "tags-activity"

    /// <summary>Health controller routes constants.</summary>
    module HealthController =
        /// <summary>Health route.</summary>
        let Health = Base + "/health-check"

    /// <summary>Exports controller routes constants.</summary>
    module ExportsController =
        /// <summary>Exports route.</summary>
        let Exports = Base + "/exports"
